console/console-init/ui/src/modules/device/dialogs/CreateDevice/AddGateway.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Title,
+  TextInput,
+  FormGroup,
+  Form,
+  Button,
+  FlexItem,
+  Flex,
+  Chip,
+  ChipGroup,
+  GridItem,
+  Grid
+} from "@patternfly/react-core";
+import { DeviceListAlert } from "modules/device/components";
+import { deviceIDRegExp } from "utils";
+
+export interface IGatewayProps {
+  addedGateways: string[];
+  removeGateway: (id: string) => void;
+  addGateway: (deviceId: string) => void;
+}
+
+export const AddGateway: React.FunctionComponent<IGatewayProps> = ({
+  addedGateways,
+  removeGateway,
+  addGateway
+}) => {
+  const [inputID, setInputID] = useState<string>("");
```

Comment:
we can initialize the inputID as undefined instaed of empty string.

Reply:
done